packages/next/build/webpack/plugins/build-manifest-plugin.ts

```diff
@@ -63,10 +87,25 @@ function getFilesArray(files: any) {
 export default class BuildManifestPlugin {
   private buildId: string
   private modern: boolean
+  private rewrites: Rewrite[]
 
-  constructor(options: { buildId: string; modern: boolean }) {
+  constructor(options: {
+    buildId: string
+    modern: boolean
+    rewrites: Rewrite[]
+  }) {
     this.buildId = options.buildId
     this.modern = options.modern
+    // TODO: should public files be sent in case they match and the
+    // rewrites chain should be broken?
+    this.rewrites = options.rewrites.map((r) => {
+      const rewrite = { ...r }
+
+      if (!rewrite.destination.startsWith('/')) {
```

Comment:
We need a comment for what this is testing. I assume it's checking if it's a page destination and not an external rewrite?

Reply:
Right, added comment for explanation